drivers/virtualbox/virtualbox_test.go

```
@@ -1,10 +1,207 @@
+func TestStateRunning(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="running"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Running, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStatePaused(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="paused"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Paused, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateSaved(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="saved"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Saved, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateStopped(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="poweroff"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateAborted(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="aborted"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateUnknown(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="whatever"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.None, machineState)
+	assert.NoError(t, err) // Really?
```

Heh, "Really" as in "Seriously, we don't return an error in this case?"  I agree it should raise an error.